services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent-document/notice-of-intent-document.service.ts

```diff
@@ -0,0 +1,304 @@
+
+@Injectable()
+export class NoticeOfIntentDocumentService {
+  private DEFAULT_RELATIONS: FindOptionsRelations<NoticeOfIntentDocument> = {
+    document: true,
+    type: true,
+  };
+
+  constructor(
+    private documentService: DocumentService,
+    private noticeOfIntentService: NoticeOfIntentService,
+    @InjectRepository(NoticeOfIntentDocument)
+    private noticeOfIntentDocumentRepository: Repository<NoticeOfIntentDocument>,
+    @InjectRepository(DocumentCode)
+    private documentCodeRepository: Repository<DocumentCode>,
+  ) {}
+
+  async attachDocument({
+    fileNumber,
+    fileName,
+    file,
+    documentType,
+    user,
+    system,
+    source = DOCUMENT_SOURCE.ALC,
+    visibilityFlags,
+  }: {
+    fileNumber: string;
+    fileName: string;
+    file: MultipartFile;
+    user: User;
+    documentType: DOCUMENT_TYPE;
+    source?: DOCUMENT_SOURCE;
+    system: DOCUMENT_SYSTEM;
+    visibilityFlags: VISIBILITY_FLAG[];
+  }) {
+    const noticeOfIntent = await this.noticeOfIntentService.getByFileNumber(
+      fileNumber,
+    );
+    const document = await this.documentService.create(
+      `noticeOfIntent/${fileNumber}`,
+      fileName,
+      file,
+      user,
+      source,
+      system,
+    );
+    const appDocument = new NoticeOfIntentDocument({
+      typeCode: documentType,
+      noticeOfIntent,
+      document,
+      visibilityFlags,
+    });
+
+    return this.noticeOfIntentDocumentRepository.save(appDocument);
+  }
+
+  async attachDocumentAsBuffer({
+    fileNumber,
+    fileName,
+    file,
+    mimeType,
+    fileSize,
+    documentType,
+    user,
+    system,
+    source = DOCUMENT_SOURCE.ALC,
+    visibilityFlags,
+  }: {
+    fileNumber: string;
+    fileName: string;
+    file: Buffer;
+    mimeType: string;
+    fileSize: number;
+    user: User;
+    documentType: DOCUMENT_TYPE;
+    source?: DOCUMENT_SOURCE;
+    system: DOCUMENT_SYSTEM;
+    visibilityFlags: VISIBILITY_FLAG[];
+  }) {
+    const noticeOfIntent = await this.noticeOfIntentService.getByFileNumber(
+      fileNumber,
+    );
+    const document = await this.documentService.createFromBuffer(
+      `noticeOfIntent/${fileNumber}`,
+      fileName,
+      file,
+      mimeType,
+      fileSize,
+      user,
+      source,
+      system,
+    );
+    const appDocument = new NoticeOfIntentDocument({
+      typeCode: documentType,
+      noticeOfIntent,
+      document,
+      visibilityFlags,
+    });
+
+    return this.noticeOfIntentDocumentRepository.save(appDocument);
+  }
+
+  async get(uuid: string) {
+    const document = await this.noticeOfIntentDocumentRepository.findOne({
+      where: {
+        uuid: uuid,
+      },
+      relations: this.DEFAULT_RELATIONS,
+    });
+    if (!document) {
+      throw new NotFoundException(`Failed to find document ${uuid}`);
+    }
+    return document;
+  }
+
+  async delete(document: NoticeOfIntentDocument) {
+    await this.noticeOfIntentDocumentRepository.remove(document);
```

Comment:
completely optional, now we know that this one could be wrapped into transaction in typeorm